develop/tutorials/articles/300-osgi-basics/03-adding-third-party-libraries-to-a-module.markdown

```diff
@@ -1,46 +1,41 @@
 # Adding Third Party Libraries to a Module [](id=adding-third-party-libraries-to-a-module)
 
 The OSGi framework is intended for modular development where you can create
-modules containing explicit information about their requirements and
-capabilities that can be used to assemble a working system. In a perfect world,
-every library would contain this information (i.e., as OSGi modules).
+multiple modules that can be composed together to build an overall application.
+In order to enable safe composition of a working system, OSGi requires all modules
+to contain explicit information about their requirements and capabilities.  Then the
+OSGi system can use this information to assemble a working system. In a perfect world,
+every existing java library would contain this information (i.e., as full-fledged OSGi bundle).
 Unfortunately, not all libraries are packaged this way. So how do you use
-them in OSGi?
+them in your OSGi module?
 
-You can add them within your module. This isn't the only way, but it's the
-method that most resembles building WAR files, so it's the one we recommend for
-beginners.
+If your third party library is not already available as a OSGi bundle, then you must find
+a way to package that third-party library in a way that is consumeable by your module.
+There are a few methods available.
 
-There are a couple different ways to add libraries to your modules:
+-   **Wrap** the third party code inside OSGi framework as a stand-alone module.
+-   **Copy** specific java packages of the libraries directly into the module.
+-   **Embed** the libraries wholesale directly into your final module and reference them in the
+    module's classpath.
 
--   **Embed** the libraries in a module folder and reference them in the
-    module's classpath. 
--   **Expand** specific parts of the libraries directly into the module.
+For those that want to avoid the details of creating or managing OSGi metadata
+which would be required if you were going to **wrap** the original library,
+we recommand that you choose one of the last two options, **copying** or **embedding**
+the required library into your module.  This method most resembles building WAR files
+(copying libraries into WEB-INF/lib), so it's the one we recommend.
 
```

Comment:
@gamerson It still seems worth mentioning how to handle resources and descriptors that might be using the same names. Did you remove the paragraphs below on purpose? Do you think the topic should be explained differently?

... Thanks